First implementation to search for shifter

CodyKank opened the pull request:
Note that back end job execution is still hardcoded for singularity arguments. This is a work in progress.

CodyKank committed:
First implementation to search for shifter

Changed how the htcondor job wrapper searches for
a container. Instead of assuming singularity
now searches through a list to be expanded
later.

Singularity is still hardcoded with arguments.

khurtado commented on files/job_wrapper.sh:
```diff
@@ -24,25 +24,51 @@ populate(){
+find_container(){
+    declare -a search_list=("singularity" "shifter")
+    declare -a module_list=(singularity tacc-singularity)
+    declare -A found_list=()
```
Note the use of associative arrays means we will need bash >= 4.x. I can't see why that would be a problem on any Linux-based OS though, considering the release date was about 10 years ago (Mac is the only OS I know of that still uses bash 3.x due to licensing issues), but we should probably put a comment at the top of the script, just in case.

CodyKank replied:
`Blue Waters` nodes have bash version: `GNU bash, version 3.2.51(1)-release (x86_64-suse-linux-gnu)` so there is an issue. Will consider other options and rework that section of code. Thanks for bringing this up.

CodyKank replied:
Resolved in 3951c0478931f9333c349efacbab5daed2833c1f

khurtado commented on files/job_wrapper.sh:
```diff
@@ -24,25 +24,51 @@ populate(){
+#         1: Couldn't find a container
+find_container(){
+    declare -a search_list=("singularity" "shifter")
+    declare -a module_list=(singularity tacc-singularity)
```
Just logged into bluewaters and got into a node interactively. I had to type `module load shifter`, so it seems that should also get in the list.

CodyKank replied:
You're absolutely right, not sure why I forgot to add that...

CodyKank replied:
Resolved in e4991c7495b8e0ca7034144a040d23d03be0a819

CodyKank committed:
Add shifter to module search list in job_wrapper

Erroneously left out shifter in module search list,
fixed incorrect usage of cntr_path in job_wrapper.

CodyKank committed:
Remove bash associative arrays.

To account for environments with bash versions > 4.0
associative arrays in job_wrapper for htcondor were
removed and replaced with simple arrays.
Adjusted call to modules within job_wrapper due to
failure on Blue Waters.
